compiler/ballerina-lang/src/main/java/org/wso2/ballerinalang/compiler/semantics/analyzer/SemanticAnalyzer.java

```diff
@@ -820,7 +820,6 @@ public void visit(BLangSimpleVariable varNode) {
                     types.isAssignable(lhsType, symTable.decimalType) ||
                     types.isAssignable(lhsType, symTable.xmlType) ||
                     types.isAssignable(lhsType, symTable.arrayType) ||
-                    types.isAssignable(lhsType, symTable.mapAnydataType) ||
```

Comment:
Don't we need to refactor the getAsRecordAndMark in TomlProvider with proper error?

Reply:
I think we can remove that method from ConfigProvider

Reply:
removed in this PR  #29381